stream-chat-android-core/src/main/java/io/getstream/chat/android/client/call/Call.kt

```diff
@@ -27,14 +27,42 @@ public interface Call<T : Any> {
      * Executes the call asynchronously, on a background thread. Safe to call from the main
      * thread. The [callback] will always be invoked on the main thread.
      */
-    public fun enqueue(callback: (Result<T>) -> Unit = {})
+    @Suppress("DeprecatedCallableAddReplaceWith", "NEWER_VERSION_IN_SINCE_KOTLIN")
+    @Deprecated(
+        level = DeprecationLevel.ERROR,
+        message = "Use the enqueue method with a Callback<T> parameter instead",
+    )
+    // Prevent usages from Kotlin, force call resolution to select the new enqueue function
+    @SinceKotlin("99999.9")
```

Comment:
Really nice

Reply:
That's an overstatement, but it's a hack that works, and I don't know of another one 😄

Reply:
I mean, super call how it works. Completely hides the method from kotlin users.